src/main/ruby/core/posix.rb

```diff
@@ -217,6 +218,45 @@ def self.unsetenv(name)
   else
     raise 'Unsupported platform'
   end
+  
+  def self.write_rope(fd, string, nonblock=false)
+    written = 0
+    Truffle::StringOperations.visit_native_bytes(string) do |pointer, length|
+      to_write = length
+      while to_write > 0
+        ret = write(fd, pointer, length)
+        if ret < 0
+          if nonblock
+            errno = Errno.nfi_errno
+            if errno == Errno::EAGAIN::Errno || errno == Errno::EWOULDBLOCK::Errno
+              raise IO::EAGAINWaitWritable
+            else
+              Errno.handle_nfi
+            end
+          else
+            Errno.handle_nfi
+          end
```

Comment:
This could be simplified with a single `Errno.handle_nfi` on this line:
```ruby
if nonblock
  if eagain
    raise
  end
end
Errno.handle_nfi

Reply:
Done.